dbms/src/TiDB/Schema/TiDBSchemaSyncer.h

```diff
@@ -110,14 +116,27 @@ struct TiDBSchemaSyncer : public SchemaSyncer
         auto getter = createSchemaGetter(keyspace_id);
 
         Int64 version = getter.getVersion();
-        if (version <= cur_version)
+        if (version == -1 && cur_version == version)
+        {
+            // Tables and databases are already tombstoned and waiting for GC.
+            return false;
+        }
```

Comment:
I think it's too implicit, the reader does not know the table's dropped in `builder.syncAllSchemas`. How about constructing a builder here and calling applyDropSchema for each table/database explicitly?

Reply:
Is there a chance that `SchemaGetter::getVersion` get `-1`, but the tables and databases metadata on TiKV does not get cleaned yet?

Reply:
> Is there a chance that `SchemaGetter::getVersion` get `-1`, but the tables and databases metadata on TiKV does not get fully cleaned yet?

I think  partial deletion of an entire keyspace is illegal.

Reply:
how do the Regions of a "deleted" keyspace get removed from TiKV? Are they removed one by one?

Reply:
@JaySon-Huang No, the keyspace range will be delete in two `UnsafeDestroyRange` request, one for transaction kv space, the other for raw kv space.

Reply:
I agree that it's better make the logic more direct, the logic now is to treat the `version=-1` as same as `version>=0` to sync schema, but return ASAP in `tryLoadSchemaDiffs` if `version is -1`. 
I wonder is it necessary to implement a branch to `syncAllSchemas` and maintain the metrics ? 🤦 /cc @JaySon-Huang @iosmanthus

Reply:
Considering of possibility inconsistency of `getVersion` and `listDBs` after `UnsafeDestroyRange`, we do not need to `listDB` or `listTables` if the schema version key not exists. It‘s neccessary to run shortcut logic in this case. We can add a new method `SchemaBuilder::dropAllSchema()` to resolve this. /cc @JaySon-Huang @iosmanthus

Reply:
LGTM